frontends/common/constantParsing.cpp

```diff
@@ -27,6 +27,10 @@ std::ostream &operator<<(std::ostream &out, const UnparsedConstant &constant) {
     return out;
 }
 
+bool operator<(const UnparsedConstant &a, const UnparsedConstant &b) {
+    return a.text < b.text || a.skip < b.skip || a.base < b.base || (!a.hasWidth && b.hasWidth);
```

Comment:
This is not a consistent operator. You can have `a < b && b < a` hold. Usually if you have a structure that just holds data and want to provide a generic comparison, the easier way is to do `std::tie(a.x, a.y, ...) < std::tie(b.x, b.y, ...)` and just let the tuple's implementation of lexicographic  comparison handle it. Of course, with C++20 we could just default such comparisons, but we still don't have it (and GCC 9 does not support it anyway).